drone/deploy.go

```diff
@@ -38,16 +38,37 @@ func deploy(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	number, err := strconv.Atoi(c.Args().Get(1))
-	if err != nil {
-		return err
-	}
 
 	client, err := newClient(c)
 	if err != nil {
 		return err
 	}
 
+	buildArg := c.Args().Get(1)
+	var number int
+	if buildArg == "last" {
+		// Fetch the build number from the last build
+		builds, err := client.BuildList(owner, name)
+		if err != nil {
+			return err
+		}
+		for _, build := range builds {
+			if build.Status == model.StatusSuccess && build.Event != model.EventDeploy {
```

Comment:
or may be simple add params for event type and branch ? so by default it uses only push event type, but if user wants it can be deploy event and pull request...? what do you think?